### PR TITLE
Security: Add renovate config in Mimir

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### Which issue(s) this PR fixes or relates to

Main Branch Dependency Updates: This ensures that dependencies for the main branch are regularly updated, keeping our project's core codebase up to date with the latest improvements and features.

Security Updates for Last 2 Release Branches: We are prioritizing security by applying updates to the last two release branches, ensuring that our users are protected against potential vulnerabilities. This is working with some delays, see [here](https://github.com/ying-jeanne/mimir/pull/46)

This configuration has been thoroughly tested on the Mimir fork, which you can find at [Mimir Fork Renovate Configuration](https://github.com/ying-jeanne/mimir/blob/main/renovate.json).

#### Checklist

- na Tests updated
- na Documentation added
- na `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: renovate.json
Comment (charleskorn) [inline renovate.json:24]: I'd suggest breaking this into separate groups per base image, so that we get one PR for Alpine updates, and another PR for Golang updates etc. 

Otherwise, they'll all be grouped together, which may not be desirable (eg. we only want to update Alpine or Golang right now, but not both).

Comment (ying-jeanne) [reply renovate.json:24]: Hello, I apologize for my delayed response. I wanted to let you know that we have received approval from the security team, and I have submitted a request to install the Renovate GitHub app on Mimir. The main reason for this is to streamline the process for updating the Dockerfile located at https://github.com/grafana/mimir/blob/main/mimir-build-image/Dockerfile. By doing this, we can reduce the number of pull requests required for updating this file. Each update currently incurs a significant one-hour build and image push, so I'm considering separating only the "golang" component from the others. This change would result in us having a total of maximum 3 builds instead of the potential maximum of 15 or a minimum of 5 builds (calculated as 5 + 4 + 3 + 2 + 1 since each merge and rebase could trigger an additional build and push). What are your thoughts on this approach?

**Comment by ying-jeanne** [comment]
We have reached the consensus with security team to install renovate app for public repo, requested installation to it-support

**Comment by ying-jeanne** [comment]
We have reached the consensus with security team to install renovate app for public repo, requested installation to it-support

File: renovate.json
Comment (charleskorn) [reply renovate.json]: How often are all of these base images updated around the same time? I'd err on the side of more builds and splitting the PRs apart.

Comment (ying-jeanne) [reply renovate.json]: I have seen it when we activate dependabot only. If it is easier to start with, I can set them separately and we regroup when the case showed up again.

**Review by aknuds1 — APPROVED** [review]
LGTM - let's see how this works out?

File: renovate.json
Comment (pracucci) [inline renovate.json:7]: Does this mean we have to update this each time we do a release? Then please add it to the release instructions at `RELEASE.md`

Comment (ying-jeanne) [reply renovate.json:7]: Yes we do at the moment.

File: RELEASE.md
Comment (aknuds1) [inline RELEASE.md:82]:
```suggestion
  - [ ] Adjust the settings in the `renovate.json` configuration on the main branch to ensure that dependency updates maintain the latest two minor versions. For instance, if the current release version is 3.0, this means keeping versions 3.0 and the latest minor version from major version 2, such as 2.10.
```

**Review by aknuds1 — APPROVED** [review]
LGTM, but I suggest a minor change in RELEASE.md.

**Pull request merged**
